plonky2/src/plonk/circuit_builder.rs

```diff
@@ -531,6 +531,14 @@ impl<F: RichField + Extendable<D>, const D: usize> CircuitBuilder<F, D> {
         }
     }
 
+    /// If `condition`, enforces that two routable `Target` values are equal, using Plonk's permutation argument.
+    pub fn assert_equal_if(&mut self, condition: Target, x: Target, y: Target) {
```

Comment:
nit: maybe `conditional_assert_eq` would be preferable? No strong opinion though